service/controller/v2/resource/chart/desired_test.go

```diff
@@ -23,19 +27,143 @@ func Test_DesiredState(t *testing.T) {
+				ChannelName:    "0-1-beta",
+				ReleaseName:    "chart-operator",
+				ReleaseVersion: "0.1.2",
+			},
+		},
+		{
+			name: "case 1: basic match with empty config map",
+			obj: &v1alpha1.ChartConfig{
+				Spec: v1alpha1.ChartConfigSpec{
+					Chart: v1alpha1.ChartConfigSpecChart{
+						Name: "chart-operator-chart",
+						ConfigMap: v1alpha1.ChartConfigSpecConfigMap{
+							Name:      "chart-operator-values-configmap",
+							Namespace: "giantswarm",
+						},
 						Channel: "0.1-beta",
 						Release: "chart-operator",
 					},
 				},
 			},
+			configMap: &apiv1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "chart-operator-values-configmap",
+					Namespace: "giantswarm",
+				},
+				Data: map[string]string{},
+			},
 			expectedState: ChartState{
-				ChartName:      "quay.io/giantswarm/chart-operator-chart",
+				ChartName:      "chart-operator-chart",
+				ChartValues:    map[string]interface{}{},
 				ChannelName:    "0.1-beta",
 				ReleaseName:    "chart-operator",
 				ReleaseVersion: "0.1.2",
 			},
 		},
+		{
+			name: "case 2: basic match with config map value",
+			obj: &v1alpha1.ChartConfig{
+				Spec: v1alpha1.ChartConfigSpec{
+					Chart: v1alpha1.ChartConfigSpecChart{
+						Name: "chart-operator-chart",
+						ConfigMap: v1alpha1.ChartConfigSpecConfigMap{
+							Name:      "chart-operator-values-configmap",
+							Namespace: "giantswarm",
+						},
+						Channel: "0-1-beta",
+						Release: "chart-operator",
+					},
+				},
+			},
+			configMap: &apiv1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "chart-operator-values-configmap",
+					Namespace: "giantswarm",
+				},
+				Data: map[string]string{
+					"values.json": `{ "test": "test" }`,
+				},
+			},
+			expectedState: ChartState{
+				ChartName: "chart-operator-chart",
+				ChartValues: map[string]interface{}{
+					"test": "test",
+				},
+				ChannelName:    "0-1-beta",
+				ReleaseName:    "chart-operator",
+				ReleaseVersion: "0.1.2",
+			},
+		},
+		{
+			name: "case 3: config map with multiple values",
+			obj: &v1alpha1.ChartConfig{
+				Spec: v1alpha1.ChartConfigSpec{
+					Chart: v1alpha1.ChartConfigSpecChart{
+						Name: "chart-operator-chart",
+						ConfigMap: v1alpha1.ChartConfigSpecConfigMap{
+							Name:      "chart-operator-values-configmap",
+							Namespace: "giantswarm",
+						},
+						Channel: "0-1-beta",
+						Release: "chart-operator",
+					},
+				},
+			},
+			configMap: &apiv1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "chart-operator-values-configmap",
+					Namespace: "giantswarm",
+				},
+				Data: map[string]string{
+					"values.json": `{ "provider": "azure", "replicas": 2 }`,
+				},
+			},
+			expectedState: ChartState{
+				ChartName: "chart-operator-chart",
+				ChartValues: map[string]interface{}{
+					"provider": "azure",
+					// Numeric values in JSON will be deserialized to a float64.
+					"replicas": float64(2),
```

Comment:
Numeric values in the JSON will be decoded to a float64. So this is what we use in the test.

We convert back to JSON when calling the Helm client. So as long as the JSON is correct then we are good.